imnodes.cpp

```diff
@@ -542,6 +546,28 @@ void BeginNodeSelection(ImNodesEditorContext& editor, const int node_idx)
         depth_stack.erase(elem);
         depth_stack.push_back(node_idx);
     }
+    // Deselect a previously-selected node
+    else if (GImNodes->MultipleSelectModifier)
+    {
+        const int* const node_ptr = editor.SelectedNodeIndices.find(node_idx);
+        editor.SelectedNodeIndices.erase(node_ptr);
+
+        // Don't allow dragging after deselecting
+        editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
+    }
+
+    // To support snapping of multiple nodes, we need to store the offset of
+    // each node in the selection to the origin of the dragged node.
+    const ImVec2 ref_origin = editor.Nodes.Pool[node_idx].Origin;
+    editor.PrimaryNodeOffset = ref_origin + GImNodes->CanvasOriginScreenSpace + editor.Panning - GImNodes->MousePos;
+
+    editor.SelectedNodeOrigins.clear();
+    for (int idx = 0; idx < editor.SelectedNodeIndices.Size; idx++)
+    {
+        const int    node = editor.SelectedNodeIndices[idx];
+        const ImVec2 node_origin = editor.Nodes.Pool[node].Origin - ref_origin;
+        editor.SelectedNodeOrigins.push_back(node_origin);
```

Comment:
Maybe the naming could be clarified here, to indicate that we're storing offsets to `PrimaryNodeOffset` -- maybe `SelectedNodeOriginOffsets`?

Reply:
Updated to SelectedNodeOffsets.